Helper/Entity/ProductHelper.php

```diff
@@ -380,11 +380,7 @@ protected function handlePrice(Product &$product, $subProducts, &$customData)
         $currencies = $store->getAvailableCurrencyCodes();
         $baseCurrencyCode = $store->getBaseCurrencyCode();
 
-        $groups = [];
-
-        if ($areCustomersGroupsEnabled) {
```

Comment:
Why did you remote this condition? It doesn't need to fetch groups when they are not enabled, no?

Reply:
If I leave this condition I have to duplicate foreach from 407th line in order to avoid problem with undefined `$groups` variable. I've added filter to speed up the code and keep properly defined `$groups` variable